metric/builds.go

```diff
@@ -59,57 +73,68 @@ func FailedBuildCount() *prometheus.CounterVec {
 			Name: "harness_ci_pipeline_execution_errors_total",
 			Help: "Total number of pipeline executions which failed due to system errors",
 		},
-		[]string{"pool_id", "os", "arch", "driver"},
+		[]string{"pool_id", "os", "arch", "driver", "distributed"},
 	)
 }
 
 // RunningCount provides metrics for number of builds currently running
-func RunningCount(instanceStore store.InstanceStore) *prometheus.GaugeVec {
+func RunningCount() *prometheus.GaugeVec {
 	return prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Name: "harness_ci_pipeline_running_executions",
 			Help: "Total number of running executions",
 		},
-		[]string{"pool_id", "os", "arch", "driver", "state"}, // state can be running, in_use, or hibernating
+		[]string{"pool_id", "os", "arch", "driver", "state", "distributed"}, // state can be running, in_use, or hibernating
 	)
 }
 
 // RunningPerAccountCount provides metrics at account level for running executions
 // This might be removed in the future as we don't want labels with high cardinality
 // We are just using two labels at the moment which should be pretty small
-func RunningPerAccountCount(instanceStore store.InstanceStore) *prometheus.GaugeVec {
+func RunningPerAccountCount() *prometheus.GaugeVec {
 	return prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Name: "harness_ci_pipeline_per_account_running_executions",
 			Help: "Total number of running executions per account",
 		},
-		[]string{"owner_id", "os"},
+		[]string{"owner_id", "os", "distributed"},
```

Comment:
+1 (applies to all metrics in this file) why do we need distributed? Only add fields to metrics which are absolutely needed and make sense to query on.
With prometheus every new entry you store here multiplies the size of the timeseries database